dynamodb/src/test/scala/zio/dynamodb/ExecutorSpec.scala

```diff
@@ -160,4 +175,149 @@ object ExecutorSpec extends DefaultRunnableSpec with DynamoDBFixtures {
+            projectionExpressionSet = Set.empty
+          )
+        )
+      )
+    )
+
+  private val batchWriteRequest = BatchWriteItem(
+    MapOfSet.empty[TableName, BatchWriteItem.Write] + (
+      (
+        TableName(mockBatches),
+        BatchWriteItem.Put(itemOne)
+      )
+    ),
+    retryPolicy = Schedule.recurs(1)
+  )
+
+  private val firstWriteRequest =
+    DynamoDBExecutorImpl.generateBatchWriteItem(
+      batchWriteRequest
+    )
+
+  private val mockedBatchGet: ULayer[DynamoDb] = DynamoDbMock
+    .BatchGetItem(
+      equalTo(firstGetRequest),
+      value(
+        ZIOAwsBatchGetItemResponse(
+          responses = Some(
+            ScalaMap(
+              mockBatches -> List(
+                ScalaMap("k1" -> ZIOAwsAttributeValue(s = Some("v1")))
+              )
+            )
+          ),
+          unprocessedKeys = Some(
+            ScalaMap(
+              mockBatches -> ZIOAwsKeysAndAttributes(
+                keys = List(ScalaMap("k1" -> ZIOAwsAttributeValue(s = Some("v2"))))
+              )
+            )
+          )
+        ).asReadOnly
+      )
+    ) ++ DynamoDbMock
+    .BatchGetItem(
+      equalTo(retryGetRequest),
+      value(
+        ZIOAwsBatchGetItemResponse(
+          responses = Some(
+            ScalaMap(
+              mockBatches -> List(
+                ScalaMap(
+                  "k1" -> ZIOAwsAttributeValue(s = Some("v2")),
+                  "k2" -> ZIOAwsAttributeValue(s = Some("v23"))
+                )
+              )
+            )
+          ),
+          unprocessedKeys = None
+        ).asReadOnly
+      )
+    )
+
+  private val batchGetSuite =
+    suite("retry batch gets")(
+      testM("should retry when there are unprocessed keys") {
+        for {
+          response <- BatchGetItem(
+                        requestItems = ScalaMap(
+                          TableName(mockBatches) -> BatchGetItem.TableGet(
+                            Set(PrimaryKey("k1" -> "v1"), PrimaryKey("k1" -> "v2")),
+                            Set.empty
+                          )
+                        ),
+                        retryPolicy = Schedule.recurs(1)
+                      ).execute
+        } yield assert(response.responses.get(TableName(mockBatches)))(
+          equalTo(Some(Set(itemOne, Item("k1" -> "v2", "k2" -> "v23"))))
+        )
+      }
+    )
+
+  private val mockedBatchWrite: ULayer[DynamoDb] = DynamoDbMock
+    .BatchWriteItem(
+      equalTo(firstWriteRequest),
+      value(
+        BatchWriteItemResponse(
+          unprocessedItems = Some(
+            ScalaMap(
+              mockBatches -> Set(
+                DynamoDBExecutorImpl.batchItemWriteToZIOAwsWriteRequest(BatchWriteItem.Put(itemOne))
+              )
+            )
+          )
+        ).asReadOnly
+      )
+    )
+    .atMost(2)
+
+  private val batchWriteSuite =
+    suite("retry batch writes")(
+      testM("should retry when there are unprocessed items") {
```

Comment:
do we have a test for "retrys" > "MAX retrys in policy " ? I think is will test the `catchSome` logic on the negative path.

Reply:
We don't have a test for this yet but I'll add some

Reply:
Actually I did have one for just the writing, I've added one success and one failure for get/write to make sure both `catchSome`'s are tested.